libtransmission/variant.c

```diff
@@ -1141,8 +1141,21 @@ tr_variantMergeDicts (tr_variant * target, const tr_variant * source)
             }
           else if (tr_variantIsList (val))
             {
-              if (tr_variantDictFind (target, key) == NULL)
-                tr_variantListCopy (tr_variantDictAddList (target, key, tr_variantListSize (val)), val);
+              tr_variant * targetList = tr_variantDictFind (target, key);
+              if (targetList == NULL)
+                {
+                  tr_variantListCopy (tr_variantDictAddList (target, key, tr_variantListSize (val)), val);
+                }
+              else
+                {
+                  // if we are processing TR_KEY_download_groups list
+                  // overwrite target with source
+                  if (key == TR_KEY_download_groups && tr_variantListSize(val) > 0)
```

Comment:
@mikedld This is the part I was talking about.

When saving the settings into the JSON file, since the possible downloadGroup is an array, without this "tweak" the array in the current settings doesn't get overridden as it should. Do you have any idea how to do this correctly without this ?